unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarAnimatorLegacy.cs

```diff
@@ -119,13 +118,20 @@ public bool Prepare(string bodyshapeId, GameObject container)
         animation.Sample();
         InitializeAvatarAudioAndParticleHandlers(animation);
 
-        if (isOwnPlayer)
+        // since the avatar can be updated when changing a wearable we shouldn't register to the update event twice
+        if (!isUpdateRegistered)
         {
-            DCLCharacterController.i.OnUpdateFinish += OnUpdateWithDeltaTime;
-        }
-        else
-        {
-            Environment.i.platform.updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, OnEventHandlerUpdate);
+            isUpdateRegistered = true;
+
+            if (isOwnPlayer)
+            {
+                DCLCharacterController.i.OnUpdateFinish += OnUpdateWithDeltaTime;
+            }
+            else
+            {
+                Environment.i.platform.updateEventHandler.AddListener(IUpdateEventHandler.EventType.Update, OnEventHandlerUpdate);
```

Comment:
to avoid subscribing twice can you do something like: `Environment.i.platform.updateEventHandler.RemoveListener(IUpdateEventHandler.EventType.Update, OnEventHandlerUpdate);` ?